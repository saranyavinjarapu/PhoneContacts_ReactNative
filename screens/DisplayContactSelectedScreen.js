import React from "react";
import { StyleSheet, View, Button } from "react-native";
import SelectedContactItem from "../components/SelectedContactItem";

const DisplayContactSelectedScreen = (props) => {
  return (
    <View style={styles.screen}>
      <SelectedContactItem
        visible={props.is_contact_selected}
        phoneContactSelected={props.selected_contact_data}
      ></SelectedContactItem>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    padding: 10,
    alignContent: "center",
    justifyContent: "center",
    alignItems: "center",
  },
});

export default DisplayContactSelectedScreen;
