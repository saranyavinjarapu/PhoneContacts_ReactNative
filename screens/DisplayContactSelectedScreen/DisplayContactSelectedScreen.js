import React from "react";
import { View, Button } from "react-native";
import styles from "./DisplayContactSelectedScreenStyles";
import SelectedContactItem from "../../components/SelectedContactItem/SelectedContactItem";

const DisplayContactSelectedScreen = (props) => {
  return (
    <View style={styles.screen}>
      <SelectedContactItem
        visible={props.is_contact_selected}
        phoneContactSelected={props.selected_contact_data}
      ></SelectedContactItem>
      <View style={styles.button}>
        <Button
          title="BACK TO HOME PAGE"
          onPress={props.returnHomePage}
        ></Button>
      </View>
    </View>
  );
};

export default DisplayContactSelectedScreen;
