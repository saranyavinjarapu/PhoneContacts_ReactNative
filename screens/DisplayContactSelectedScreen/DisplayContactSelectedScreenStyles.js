import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  screen: {
    padding: 10,
    alignContent: "center",
    justifyContent: "center",
    alignItems: "center",
  },
  button: {
    width: 250,
    padding: 10,
    marginTop: 350,
  },
});

export default styles;
