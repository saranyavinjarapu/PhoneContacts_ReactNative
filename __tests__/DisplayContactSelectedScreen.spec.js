import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import DisplayContactSelectedScreen from "../screens/DisplayContactSelectedScreen";

describe("Display Contact Selected Screen Renders Correctly", () => {
  it("renders correctly", () => {
    const snap = render(<DisplayContactSelectedScreen />).toJSON();
    expect(snap).toMatchSnapshot();
  });
  it("Back to Home Page Button Works", () => {
    const { queryByText } = render(<DisplayContactSelectedScreen />);
    const sendHandler = jest.fn();
    const button = queryByText("BACK TO HOME PAGE");
    expect(button).toBeTruthy();
  });
});
