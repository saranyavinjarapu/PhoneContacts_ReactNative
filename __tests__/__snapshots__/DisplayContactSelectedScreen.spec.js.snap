// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Contact Selected Screen Renders Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 15,
          "justifyContent": "center",
          "marginVertical": 100,
          "maxWidth": "80%",
          "padding": 90,
          "paddingLeft": 130,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.26,
          "shadowRadius": 6,
          "textAlign": "center",
          "width": "80%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 350,
        "padding": 10,
        "width": 250,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          BACK TO HOME PAGE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
